### ISSUE TITLE
[Question] Testing multi-node set up

### ISSUE THREAD

**Issue by gohilankit** [kubernetes-csi/csi-test#294]
Is it possible to run sanity tests on a multi-node k8s cluster? I don't see any way to specify node endpoint as a runtime argument.

When I currently run the tests on a multi-node cluster, I get the error `unknown service csi.v1.Node`. How can this be resolved?

```
root@4217a296e769577a9da4a7f751f2ba4a [ ~/csi-test/cmd/csi-sanity ]# ./csi-sanity --csi.endpoint=unix:///var/lib/csi/sockets/pluginproxy/csi.sock --ginkgo.failFast --ginkgo.focus="CreateVolume"
Running Suite: CSI Driver Test Suite
====================================
Random Seed: 1600472653
Will run 11 of 77 specs

SSSSSS••STEP: reusing connection to CSI driver at unix:///var/lib/csi/sockets/pluginproxy/csi.vsphere.vmware.com/csi.sock
STEP: creating mount and staging directories
STEP: creating a volume
cleanup volumes: deleting 348435d8-8f1a-4644-adac-81a2a31cec3a
cleanup volumes: NodeUnpublishVolume failed: rpc error: code = Unimplemented desc = unknown service csi.v1.Node

------------------------------
• Failure in Spec Teardown (AfterEach) [4.086 seconds]
Controller Service [Controller Server]
/root/csi-test/pkg/sanity/tests.go:44
  CreateVolume [AfterEach]
  /root/csi-test/pkg/sanity/controller.go:368
    should return appropriate values SingleNodeWriter NoCapacity
    /root/csi-test/pkg/sanity/controller.go:408

    recorded 1 failure(s)

    /root/go/pkg/mod/github.com/onsi/ginkgo@v1.10.3/internal/leafnodes/runner.go:113
```

Also, I noticed as part of clean up after each test, the delete volume is calling NodeUnpublishVolume & NodeUnstageVolume.
I don't think they need to be invoked after running test suite for say, CreateVolume?
https://github.com/kubernetes-csi/csi-test/blob/0ddc0e0dd7df52b42db71e48c890fd3c254add10/pkg/sanity/resources.go#L262

**Comment by pohly** [comment]
> Is it possible to run sanity tests on a multi-node k8s cluster? I don't see any way to specify node endpoint as a runtime argument.

The `--csi.endpoint` is for node and controller unless the controller is specified separately.

> `unknown service csi.v1.Node`

The endpoint that you connected to doesn't implement the CSI node API.

> Also, I noticed as part of clean up after each test, the delete volume is calling NodeUnpublishVolume & NodeUnstageVolume.
> I don't think they need to be invoked after running test suite for say, CreateVolume?

True. When testing just the controller part of the CSI driver that won't work.

Which version of csi-sanity are you using? There have been some changes around this in v4.x

**Comment by gohilankit** [comment]
> Which version of csi-sanity are you using? There have been some changes around this in v4.x

I'm using the latest(build from master branch). I suspect 4.0.1 that was released couple of weeks back will have the similar behavior.



> The endpoint that you connected to doesn't implement the CSI node API.

I now used both --csi.endpoint & --csi.controllerendpoint, and now able to access Node service. Hence, closing the issue for now.
But I still feel that Node endpoints should not be invoked while testing just the CSI controller RPCs.
